sqlglot/dialects/tsql.py

```diff
@@ -1,14 +1,68 @@
 from sqlglot import exp
 from sqlglot.dialects.dialect import Dialect
 from sqlglot.generator import Generator
+from sqlglot.helper import list_get
 from sqlglot.parser import Parser
+from sqlglot.time import format_time
 from sqlglot.tokens import Tokenizer, TokenType
 
 
+def tsql_format_time_lambda(exp_class, mapping=None, default=None):
```

Comment:
Why can't we use format time lambda here?

Reply:
TSQL has inversted arguments and the arguments are not tokanized correctly.
The first argument is tokenized as column value, passing that to the regular format time lambda causes unexpected behaviour where the argument is not time_mapped.

Reply:
Ok, I'll check it more thoroughly later then and let you know if I can come up with an alternative.